frontend/src/auth/auth.js

```diff
@@ -0,0 +1,41 @@
+import _ from 'lodash';
+import jwt from 'jwt-decode';
+
+class Auth {
+
+  constructor() {
+    // delay import of 'auth0-lock' because of dependency on 'packageify'
+    let Auth0Lock = require('auth0-lock');
+    this.auth0Lock = new Auth0Lock('7mdQiLLeKFZ5TlYPsEDOUKGavNWFzWSy', 'turtle.eu.auth0.com');
+    this.options = {
+      authParams: {
+        scope: 'openid name email user_id nickname picture'
+      },
+      closable: false
+    };
+  }
+
+  token() {
+    // https://auth0.com/docs/libraries/lock/types-of-applications#single-page-app
+    return new Promise((resolve, reject) => {
+      let token = localStorage.getItem('userToken');
+      if (token && jwt(token).exp < Date.now()) {
```

Comment:
I am not completely sure why `jwt(token).exp` should be less than `Date.now()`; I would think it should be the other way around. If how it is written now is in fact how should be, then great.